app/src/main/java/com/cornellappdev/transit/ui/components/SearchSuggestions.kt

```diff
@@ -35,13 +40,16 @@ fun SearchSuggestions(
             buttonText = "Add",
             onClick = onFavoriteAdd
         )
-        favorites.forEach {
+        favorites.take(minOf(5, favorites.size)).forEach {
             MenuItem(
                 type = it.type,
                 label = it.name,
                 sublabel = it.subLabel,
                 onClick = {
-                })
+                    homeViewModel.addRecent(it)
+                    navController.navigate("route/${it.name}")
+                }
+            )
```

Comment:
A few subtle problems with this:

We should never be pipelining `homeViewModel` into a component. This is a pretty common error known as "prop drilling." The general idea is that components should only provide an interface for data in / events out, and not be able to see anything about the viewmodel.

So, you should replace this with some `onStopPressed: (Place) -> Unit` (or whatever else you need) and just call that in the component here. Then, `HomeScreen` which calls this component can implement this onClick. That way, only the screen is directly interacting with the home view model. 

Ideally, you'd actually want the home screen to go one step even further and call a new `fun onStopPressed(...)` in `HomeViewModel` you make, so that even the home screen doesn't care about the implementation. This takes ALL the logic and moves it into the VM, away from ALL UI. I just don't know if you can access the nav controller from the VM? If not, I guess leaving in the screen is OK.

Reply:
Please fix the same occurrence on lines 66-67!

Reply:
Ultimately, there should be no trace of `homeViewModel` in this file.